src/hooks/twilio.ts

```diff
@@ -186,15 +188,16 @@ export const useVideoRoomState = ({
 
   const disconnect = useCallback(() => {
     setRoom((currentRoom) => {
-      if (currentRoom && currentRoom.localParticipant.state === "connected") {
-        currentRoom.localParticipant.tracks.forEach((trackPublication) => {
-          (trackPublication.track as LocalVideoTrack).stop();
-        });
-        currentRoom.disconnect();
-        return undefined;
-      } else {
+      if (!currentRoom || currentRoom?.localParticipant?.state !== "connected")
         return currentRoom;
```

Comment:
Thinking about this more, I think we can drop the `!currentRoom`, since the check uses the 'safe accessors' anyway.

I believe this will have the same effect, yet be slightly less cognitively burdensome to reason about:

```suggestion
      if (currentRoom?.localParticipant?.state !== "connected")
        return currentRoom;
```

Reply:
done in https://github.com/sparkletown/sparkle/pull/1299/commits/861319deb84fb4a48793c0573385068392dc8657